packages/editor/src/plugins/text/components/suggestions.tsx

```diff
@@ -1,65 +1,116 @@
-import { cn } from '@serlo/frontend/src/helper/cn'
+import React, { useEffect, useRef } from 'react'
 
-import type { SuggestionOption } from '../hooks/use-suggestions'
+import { SuggestionItem } from './suggestion-item'
+import { SuggestionOption } from '../hooks/use-suggestions'
 
 interface SuggestionsProps {
-  options: SuggestionOption[]
-  suggestionsRef: React.MutableRefObject<HTMLDivElement | null>
-  selected: number
-  onMouseDown: (pluginType: string) => void
-  onMouseMove: (index: number) => void
+  basicOptions: SuggestionOption[]
+  interactiveOptions: SuggestionOption[]
+  searchString: string
+  setSearchString: (searchString: string) => void
+  itemRefs: React.MutableRefObject<(HTMLButtonElement | null)[]>
+  currentlyFocusedItem: number
+  setCurrentlyFocusedItem: (index: number) => void
+  insertSelectedPlugin: (pluginType: EditorPluginType | string) => void
 }
 
-export const Suggestions = ({
-  options,
-  suggestionsRef,
-  selected,
-  onMouseDown,
-  onMouseMove,
-}: SuggestionsProps) => {
+export function Suggestions(props: SuggestionsProps) {
+  const {
+    basicOptions,
+    interactiveOptions,
+    searchString,
+    setSearchString,
+    itemRefs,
+    currentlyFocusedItem,
+    setCurrentlyFocusedItem,
+    insertSelectedPlugin,
+  } = props
+
   const editorStrings = useEditorStrings()
+  const searchInputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    const focusTimeout = setTimeout(() => {
+      searchInputRef.current?.focus()
+    }, 0)
+    return () => clearTimeout(focusTimeout)
+  }, [])
+
+  const getTooltipPosition = (index: number): 'right' | 'left' | undefined => {
+    return index % 5 === 0 ? 'right' : index % 5 === 4 ? 'left' : undefined
+  }
+
+  const handleItemClick = (
+    event: React.MouseEvent<HTMLButtonElement>,
+    pluginType: EditorPluginType | string
+  ) => {
+    event.preventDefault()
+    insertSelectedPlugin(pluginType)
+  }
+
+  const handleItemFocus = (index: number) => {
+    setCurrentlyFocusedItem(index)
+  }
+
+  const handleItemBlur = () => {
+    setCurrentlyFocusedItem(-1)
+  }
 
-  if (options.length === 0) {
-    return <div>{editorStrings.plugins.text.noItemsFound}</div>
+  const renderSuggestions = (
+    options: SuggestionOption[],
+    offset: number = 0
+  ) => {
+    return options.map((item, index) => {
+      const currentIndex = index + offset
+      return (
+        <SuggestionItem
+          key={currentIndex}
+          option={item}
+          selected={currentIndex === currentlyFocusedItem}
+          ref={(el) => (itemRefs.current[currentIndex] = el)}
+          onClick={(event) => handleItemClick(event, item.pluginType)}
+          onFocus={() => handleItemFocus(currentIndex)}
+          onBlur={handleItemBlur}
+          onMouseEnter={() => handleItemFocus(currentIndex)}
+          onMouseLeave={() => {
+            handleItemBlur()
+            itemRefs.current[currentIndex]?.blur()
+          }}
```

Comment:
It would be nice to do the same / a similar thing on arrow button click. Otherwise, it can happen that two tooltips are displayed one over the other.

<img width="557" alt="Screenshot 2024-08-06 at 13 45 27" src="https://github.com/user-attachments/assets/da2495ca-e4e8-474d-9b30-746c86e85009">

Reply:
This seems quite tricky, will have to investigate a bit more. Will work on fixing the e2e tests first

Reply:
(interesting edge cases when I move the focus with the keyboard, trigger a scroll and that triggers a mouseenter on a previous item 😶‍🌫️)

Reply:
I wonder if decoupling mouse interaction and keyboard interaction would be easier/better here. 
Hovering an item would not move the focus to that item but show the hover state as before. Focus would then usually stay on the input field for mouse users. You can still start filtering by typing at any point.
Double tooltips would be a (maybe ignorable) edge case.
What do you think?

Reply:
I think the reason I restored the mouse interaction and focus change was for the case where the user hovers the item with the mouse and hits the enter key, it should insert the hovered plugin

Reply:
Probably a good idea to discuss this on a call to see what the best case behavior is

Reply:
> I think the reason I restored the mouse interaction and focus change was for the case where the user hovers the item with the mouse and hits the enter key, it should insert the hovered plugin

For that you could also query for items with `:hover` on enter.